src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        array[size] = value;
+        size++;
+        growIfRequared();
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexData(index);
+        growIfRequared();
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        growIfRequared();
```

Comment:
```suggestion
```
addAll is functionally equivalent to calling add in a loop, nothing else is needed